src/main/java/codeu/model/store/basic/ConversationStore.java

```diff
@@ -72,6 +73,12 @@ private ConversationStore(PersistentStorageAgent persistentStorageAgent) {
   /** Add a new conversation to the current set of conversations known to the application. */
   public void addConversation(Conversation conversation) {
     conversations.add(conversation);
+    conversations.sort(Conversation.conversationComparator);
```

Comment:
Lets think about efficiency here a bit. 

Problem: Every time we add or update a conversation we are resorting our list. So say the list is completely unsorted when we load it from the data story in ServerStartupListener. The first time we add a conversation we will sort the whole list and have a run time of O(nlogn). Everytime after that, when adding a conversation we will probably take O(logn) to find the right spot and add it. Our runtime now is O(nlogn + xlogn) where x is the number of times we add a conversation. If x is at any point bigger than n we will have a less efficient algorithm than when we started! (granted our list is growing for n is not a static number)

Solution: Instead of sorting everytime, we should sort one time when we need this list in the right order. So in the admin page when we are trying to get the most recent conversation. 

If unfamiliar with big o notation: https://rob-bell.net/2009/06/a-beginners-guide-to-big-o-notation/
If you want me to explain this more just lmk.

Reply:
Also this makes the code more readable as well. When sorting the current way you are doing it, people will read the "getLastConversation" method and it will make little sense to them because they dont know whats so important about the last item. They will have to read more to realize that the list is sorted. I'm sure many people will assume its sorted but if at any point your sorting code accidentally gets deleted then we are in trouble!

Reply:
done